lang/cs/Org.Apache.REEF.Tests/Functional/IMRU/TestExceptionInResultHandlerDispose.cs

```diff
@@ -0,0 +1,158 @@
+            TestBroadCastAndReduce(false, numTasks, chunkSize, dims, iterations, mapperMemory, updateTaskMemory, numberOfRetryInRecovery, testFolder);
+            string[] lines = ReadLogFile(DriverStdout, "driver", testFolder, 120);
+            var completedTaskCount = GetMessageCount(lines, "Received ICompletedTask");
+            var failedEvaluatorCount = GetMessageCount(lines, FailedEvaluatorMessage);
+            var failedTaskCount = GetMessageCount(lines, FailedTaskMessage);
+            var jobSuccess = GetMessageCount(lines, IMRUDriver<int[], int[], int[], int[]>.DoneActionPrefix);
+
+            // Master evaluator will fail after master task is completed. Depending on how quick the driver dispose contexts after the master task complete,
+            // driver may or may not receive the IFailedEvalautor event. 
+            Assert.True(failedEvaluatorCount <= 1);
+
+            // Scenario1: Driver receives FailedEvaluator caused by dispose of a completed task after all the tasks have been competed. 
+            //            FailedEvaluator event will be ignored.
+            // Scenario2: Driver receives FailedEvaluator caused by dispose of master task before all the tasks have been competed.
+            //            Driver will send close event to the rest of the running tasks and enter shutdown state 
+            //            During this process, some tasks can still complete and some may fail due to communication error
+            //            As evaluator failure happens in finally block, therefore either ICompletedTask or IFailedTask event should be sent before it.
+            //            Considering once maser is done, rest of the contexts will be disposed, we have 
+            //            numTasks >= completedTask# + FailedTask#
+            Assert.True(numTasks >= completedTaskCount + failedTaskCount);
+
+            // As update task completion happens before update evaluator failure caused by dispose, eventually job should succeeds
+            Assert.Equal(1, jobSuccess);
+
+            CleanUp(testFolder);
+        }
+
+        /// <summary>
+        /// This test is for the normal scenarios of IMRUDriver and IMRUTasks on yarn
+        /// </summary>
+        [Fact(Skip = "Requires Yarn")]
+        public void TestExceptionInResultHandlerDisposerOnYarn()
+        {
+            int chunkSize = 2;
+            int dims = 10;
+            int iterations = 10;
+            int mapperMemory = 5120;
+            int updateTaskMemory = 5120;
+            int numTasks = 4;
+            int numberOfRetryInRecovery = 4;
+            TestBroadCastAndReduce(false, numTasks, chunkSize, dims, iterations, mapperMemory, updateTaskMemory, numberOfRetryInRecovery);
+        }
+
+        /// <summary>
+        /// This method defines event handlers for driver. As default, it uses all the handlers defined in IMRUDriver.Logger.Log(Level.Info, "Simulate exception in Resu
```

Comment:
`Logger.Log(Level.Info, "Simulate exception in Resu` seems to be an accidental paste